client/src/modules/email-report/email-report.routes.js

```diff
@@ -0,0 +1,81 @@
+angular.module('bhima.routes')
+  .config(['$stateProvider', function ($stateProvider) {
+
+    $stateProvider
+      .state('emailreport', {
+        url: '/email-report/',
+        controller: 'EmailReportController as EmailReportCtrl',
+        templateUrl: 'modules/email-report/index.html',
+        params: {
+          data: null,
```

Comment:
If you change this line to:
```js
params : {
  data : {},
}
```

This will make data default to an object if it is not defined.  Then you will not need all the `try/catch` statements in your controllers.